### ISSUE TITLE
Enhancing the single receiver control - Enabling Autocomplete

### ISSUE THREAD

**Issue by Patowhiz** [africanmathsinitiative/R-Instat#5506]
**Is your feature request related to a problem? Please describe.**
Currently we can't type into the single receiver control. A user has to use the selector to get the set the selected variable to use. 

**Describe the solution you'd like**
We would like to enable editing of the single receiver. The user will be able to type and get autocomplete suggestions, or to select a list of available variables without necessarily going to the selector.  This variables will come from the selector (which is also responsible for determining the selected dataframe).

**Describe alternatives you've considered**
Several alternatives were suggested in issue #5480. Sample screenshots of how this will work can be found on the same issue.

**Additional context**
I'm thinking of using the combobox for this feature rather than a textbox. The combobox already has in built features that can help us easily achieve this functionality.

**Comment by rdstern** [comment]
An alternative - or additional option, - assuming we are keeping the selector for now, is to be able to limit the variables shown in the data selector.  We already have 4 options with the right-click, so these could be added?  There is also a special right-click in the receivers too.

**Comment by Patowhiz** [comment]
@rdstern yes I get your point. Since @dannyparsons suggested we don't do an overhaul of the design. I was thinking of doing this in phases. I'm suggesting we first start by making the receiver editable and with autocomplete features, and also make it be a combobox that has variable which the user can select. These variables will come from the selector. That means, the receiver only shows variables that the selector has. Of course there will be several exceptions like, if the receiver expects a specific column type, so automatically it will filter those variables that match it's column type. So to reinforce what I have in mind, the receiver will be getting it's values from the selector then using them internally to do it's best job(Which is selection of variables).

We can have a separate issue on the selector. To address the limits and the groupings. And of course, since in theory the selector will be the source of the variables, any changes to it will be picked by this enhanced receiver.

Then we can go further and add the right clicks etc, as we investigate further as how users react to this or how it's improving usability.

If it turns out that we no longer need a selector that shows variables, I'm sure it won't be such a very difficult tasks to merge some of it's features into the receivers.

@rdstern what are your thoughts on this? I'm happy to get more suggestions into this.
I can quickly create the different issues and have them relate to this.

**Comment by Patowhiz** [comment]
@rdstern what would be going into the receiver's special right click?

**Comment by dannyparsons** [comment]
Let's please keep one idea per issue. Related discussions on grouping variables should be in a separate issue so this can focus on the idea of typing in the receiver.

**Comment by shadrackkibet** [comment]
I like this feature. This will really improve user experience. I often see users trying to type into the receiver a lot especially for the single receivers. Having this feature will allow them to do so. I think this is an example of a feature that will really have a lot of impact.

**Comment by rdstern** [comment]
I use this example to propose a dangerous suggestion, namely that one option on the right-click in a single receiver could be to "Allow typing" or some such phrase.

ggplot(aes(x = reorder(package, downloads), y = downloads)) +...

So, in the x receiver for a plot we would type the bits in bold **_reorder(_**variable**_,_**another variable**_)_**.  We would not have to type the variable names, but could still click in the data selector.  

This is dangerous, because we could easily mis-type the text - but it would dramatically improve the power of the dialogues - they would each have this "halfway" element to them.

**Comment by Patowhiz** [comment]
We discussed some of the features with @rdstern during the ICPAC workshop, so I will be implementing this for demonstration. In addition, during the workshop I saw how users struggled with changing the focus of the receiver just to select variables from the selector. This is not a high priority so will be doing it on the side though I think it would really help with simplifying many dialogs for the users
